component/instances.jsonnet

```diff
@@ -0,0 +1,38 @@
+local com = import 'lib/commodore.libjsonnet';
+local kap = import 'lib/kapitan.libjsonnet';
+local kube = import 'lib/kube.libjsonnet';
+
+local inv = kap.inventory();
+local params = inv.parameters.argocd;
+
+// TODO: move this to commodore.libjsonnet?
+local namespacedName(name, namespace='') = {
+  local namespacedName = std.splitLimit(name, '/', 1),
+  local ns = if namespace != '' then namespace else error 'missing mandatory namespace in "%s"' % name,
+  namespace: if std.length(namespacedName) > 1 then namespacedName[0] else ns,
+  name: if std.length(namespacedName) > 1 then namespacedName[1] else namespacedName[0],
+};
+
+local ArgoCD(name) =
+  local n = namespacedName(name);
+  kube._Object('argoproj.io/v1beta1', 'ArgoCD', n.name) {
+    metadata+: {
+      namespace: n.namespace,
```

Comment:
It might make sense to prevent users of the component from creating multiple ArgoCD instances in the same namespace since that probably wouldn't end well.

Could be done with something like

```jsonnet
local argocds = std.foldl(
  function(st, k)
    local ns = namespacedName(k).namespace;
    if std.setMember(ns, st.seen) then
      error "Component argocd doesn't support deploying multiple ArgoCD instances in a single namespace"
    else
      st { seen+: [ ns ] },
  std.objectFields(params.argocds),
  {
    seen: [],
    argocds: params.argocds,
  }
).argocds;
```